app/controllers/concerns/lock_ingestible_concern.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+# Logic to lock Ingestible object, to be included in related controllers
+module LockIngestibleConcern
+  extend ActiveSupport::Concern
+
+  def try_to_lock_ingestible
+    return if @ingestible.obtain_lock(current_user)
```

Comment:
This method will prolong lock (for next 15 minutes), if user already had lock on ingestible